examples/temporal_graph_learning/gen_graph.py

```diff
@@ -0,0 +1,123 @@
+import pandas as pd
+import numpy as np
+import pyarrow.parquet as pq 
+import pyarrow as pa
+
+import torch
+import dgl
+from dgl import AddReverse, Compose, ToSimple
+
+def create_dgl_graph():
+    # load dataset
+    data = 'MAG'
+    df = pd.read_csv('./DATA/{}/edges.csv'.format(data))
+    num_nodes = max(int(df['src'].max()), int(df['dst'].max())) + 1
+    print('num_nodes: ', num_nodes)
+    
+    # Extract graph structure 
+    # We only consider edge types from 100 to 109
+    timestamp = df['time'].values
```

Comment:
the timestamp is on the edges. is this MAG or GDELT?

Reply:
instead of filtering out edges, you can potentially discretize the timestamps. For example, putting 1-10 as timestamp 1, 11-20 as timestamp 2, etc.

Reply:
Yes, timestamp is on edges.

For this example, since the original MAG dataset is very large, I decided to only consider 10 timestamps instead of discretizing timestamp and using the full graph. 

When using only 10 timestamps, the DGL graph is already 100GB. I tried discretizing the timestamps, it takes too much time and the graph is too big. I think it is better to keep example data small so that user can easily test them.